followthemoney/model.py

```diff
@@ -62,49 +64,59 @@ def get_type_schemata(self, type_):
                     schemata.add(schema)
         return schemata
 
-    def make_mapping(self, mapping, key_prefix=None):
+    def make_mapping(self, mapping: Dict[str, Any], key_prefix=None):
         """Parse a mapping that applies (tabular) source data to the model."""
         return QueryMapping(self, mapping, key_prefix=key_prefix)
 
-    def map_entities(self, mapping, key_prefix=None):
+    def map_entities(self, mapping: Dict[str, Any], key_prefix=None):
         """Given a mapping, yield a series of entities from the data source."""
-        mapping = self.make_mapping(mapping, key_prefix=key_prefix)
-        for record in mapping.source.records:
-            for entity in mapping.map(record).values():
+        q_mapping = self.make_mapping(mapping, key_prefix=key_prefix)
+        for record in q_mapping.source.records:
+            for entity in q_mapping.map(record).values():
                 yield entity
 
-    def common_schema(self, left, right):
+    def common_schema(self, left: Union[str, Schema], right: Union[str, Schema]) -> Schema:
         """Select the most narrow of two schemata.
 
         When indexing data from a dataset, an entity may be declared as a
         LegalEntity in one query, and as a Person in another. This function
         will select the most specific of two schemata offered. In the example,
         that would be Person.
         """
-        left = self.get(left) or self.get(right)
-        right = self.get(right) or self.get(left)
-        if left.is_a(right):
-            return left
-        if right.is_a(left):
-            return right
-        common = list(left.schemata.intersection(right.schemata))
+        left_schema = self.get(left) or self.get(right)
+        if left_schema is None:
```

Comment:
Behaviour change, the previous was to return the opposing side's schema, not throw. Can we make this a separate PR?

Reply:
Sure, but the current code fails at runtime if either `left` or `right` are `None`: 
https://github.com/alephdata/followthemoney/blob/91e56b450a9c1bc29221e40cfc2512dfd1f97dfd/followthemoney/model.py#L86
https://github.com/alephdata/followthemoney/blob/91e56b450a9c1bc29221e40cfc2512dfd1f97dfd/followthemoney/model.py#L88